python/cuml/preprocessing/model_selection.py

```diff
@@ -161,6 +163,12 @@ def train_test_split(
     x_numba = False
     y_numba = False
 
+    if cuda.devicearray.is_cuda_ndarray(X):
```

Comment:
how about just assigning this at the beginning on line 163 rather than assign and change: `x_numba = is_cuda_array(x)`